File: app/schemas/restore.py
from __future__ import annotations
from datetime import datetime
from pydantic import BaseModel


class RestoreRequest(BaseModel):
    backup_record_id: int
    target_connection_id: int


class RestoreRunResponse(BaseModel):
    record_id: int
    status: str


class RestoreOut(BaseModel):
    id: int
    backup_record_id: int
    target_connection_id: int
    status: str
    error: str | None = None
    started_at: datetime
    finished_at: datetime | None = None
    duration_ms: int | None = None

    model_config = {"from_attributes": True}

File: frontend/src/api/restore.ts
import client from './client'

export interface Restore {
  id: number
  backup_record_id: number
  target_connection_id: number
  status: string
  error: string | null
  started_at: string
  finished_at: string | null
  duration_ms: number | null
}

export const runRestore = (backup_record_id: number, target_connection_id: number) =>
  client.post<{ record_id: number; status: string }>('/restore', { backup_record_id, target_connection_id })
export const listRestores = () => client.get<Restore[]>('/restore')
export const cancelRestore = (id: number) => client.post(`/restore/${id}/cancel`)
export const eventsUrl = (id: number) => `/api/v1/restore/${id}/events`
